packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapPolylineController.m

```diff
@@ -95,6 +99,14 @@ - (void)interpretPolylineOptions:(NSDictionary *)data
   if (geodesic && geodesic != (id)[NSNull null]) {
     [self setGeodesic:geodesic.boolValue];
   }
+
+  NSArray *patterns = data[@"pattern"];
+  if (patterns && patterns != (id)[NSNull null]) {
```

Comment:
All of the code above here is incorrect, and this shouldn't replicate that incorrect pattern; assigning a `NSNull*` to a variable of type `NSArray*` is a type violation. The cast to `(id)` here is a workaround for the compiler calling out that problem.

The better way to handle this would be to make a static helper function:
```objc
/**
 * Returns dict[key], or nil if dict[key] is NSNull.
 */
id GetValueOrNil(NSDictionary *dict, NSString *key) {
  id value = dict[key];
  return value == [NSNull null] ? nil : value;
}
```

The this code becomes:
```objc
NSArray *patterns = GetValueOrNil(data, @"pattern");
if (patterns) {
  ...
```

(If you want to update the code above to use the helper too, that would be great.)